miio/protocol.py

```diff
@@ -151,7 +151,7 @@ def _decode(self, obj, context, path):
         try:
             # pp(context)
             decrypted = Utils.decrypt(obj, context['_']['token'])
-            decrypted = decrypted.rstrip(b"\x00")
+            decrypted = decrypted[:decrypted.rfind(b"\x00")]
```

Comment:
Your solution isn't accurate. rfind returns -1 if there is no \x00 at all. decrypted will be shortened by one character in this case. The previous code didn't touch the payload.

```python
>>> decrypted = "asdfg"
>>> decrypted.rstrip("f")
'asdfg'
>>> decrypted[:decrypted.rfind("f")]
'asd'
>>> decrypted[:decrypted.rfind("h")]
'asdf'
```

Reply:
Correct. There is already a fixup for a different invalid json string scenario in that method so I'll extend that to add this special case.

Reply:
You didn't get my point. Your code introduces a new case. If 0x00 isn't present the json will be destroyed now because the last char (`}`) will be removed.

Reply:
Sorry, I wasn't clear enough in my comment. I'll revert this change and instead provide a similar way to fix the decrypted data like is done for the powerstrip edge case.